ththe azurerm_cost_management_export_resource_group has been deprecated in favour of azurerm_resource_group_cost_management_export, azurerm_subscription_cost_management_export (#14140)
hashicorp/terraform-provider-azurerm: website/docs/r/subscription_cost_management_export.html.markdown

---
subcategory: "Cost Management"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_subscription_cost_management_export"
description: |-
  Manages an Azure Cost Management Export for a Subscription.
---

# azurerm_subscription_cost_management_export

> Manages an Azure Cost Management Export for a Subscription.

Reviewer: ```suggestion
Manages a Cost Management Export for a Subscription.
```

## Example Usage

```hcl
data "azurerm_subscription" "example" {}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                = "example-storage-account"
  resource_group_name = azurerm_resource_group.example.name

  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

> resource "azurerm_resource_group_cost_management_export" "example" {

Reviewer: this is for a subscription

```suggestion
resource "azurerm_subscription_cost_management_export" "example" {
```

  name                    = "example"
  subscription_id         = azurerm_subscription.example.id
  recurrence_type         = "Monthly"
  recurrence_period_start = "2020-08-18T00:00:00Z"
  recurrence_period_end   = "2020-09-18T00:00:00Z"

  export_data_storage_location {
    storage_account_id = azurerm_storage_account.example.id

>     container_name     = "examplecontainer"

Reviewer: presumably we should create a container here too?

Author: Azure creates the container if it doesn't exist, but I agree that we should explicitly create it with Terraform to avoid the creation of unmanaged resources.

    root_folder_path   = "/root/updated"
  }

  export_data_definition {
    type       = "Usage"
    time_frame = "WeekToDate"
  }
}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) Specifies the name of the Cost Management Export. Changing this forces a new resource to be created.

* `subscription_id` - (Required) The id of the subscription on which to create an export.

* `recurrence_type` - (Required) How often the requested information will be exported. Valid values include `Annually`, `Daily`, `Monthly`, `Weekly`.

> * `recurrence_period_start` - (Required) The date the export will start capturing information.
> 
> * `recurrence_period_end` - (Required) The date the export will stop capturing information.

Reviewer 2: ```suggestion
* `recurrence_period_start_time` - (Required) The date the export will start capturing information.

* `recurrence_period_end_time` - (Required) The date the export will stop capturing information.
```

> * `delivery_info` - (Required) A `delivery_info` block as defined below.

Reviewer 2: and here

* `query` - (Required) A `query` block as defined below.

* `active` - (Optional) Is the cost management export active? Default is `true`.

---

A `export_data_storage_location` block supports the following:

* `storage_account_id` - (Required) The storage account id where exports will be delivered.

* `container_name` - (Required) The name of the container where exports will be uploaded.

* `root_folder_path` - (Required) The path of the directory where exports will be uploaded.

---

> A `export_data_definition` block supports the following:

Reviewer 2: would options make more sense herE?
```suggestion
A `export_data_options` block supports the following:
```

* `type` - (Required) The type of the query.

* `time_frame` - (Required) The time frame for pulling data for the query. If custom, then a specific time period must be provided. Possible values include: `WeekToDate`, `MonthToDate`, `BillingMonthToDate`, `TheLastWeek`, `TheLastMonth`, `TheLastBillingMonth`, `Custom`.

## Attributes Reference

In addition to the Arguments listed above - the following Attributes are exported: 

> * `id` - The ID of the Cost Management.
> 
> ## Timeouts
> 
> The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:
> 
> * `create` - (Defaults to 30 minutes) Used when creating the Cost Management.
> * `read` - (Defaults to 5 minutes) Used when retrieving the Cost Management.
> * `update` - (Defaults to 30 minutes) Used when updating the Cost Management.
> * `delete` - (Defaults to 30 minutes) Used when deleting the Cost Management.
> 
> ## Import
> 
> Cost Managements can be imported using the `resource id`, e.g.

Reviewer: ```suggestion
* `id` - The ID of the Cost Management Export for this Subscription.

## Timeouts

The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/docs/configuration/resources.html#timeouts) for certain actions:

* `create` - (Defaults to 30 minutes) Used when creating the Subscription Cost Management Export.
* `read` - (Defaults to 5 minutes) Used when retrieving the Subscription Cost Management Export.
* `update` - (Defaults to 30 minutes) Used when updating the Subscription Cost Management Export.
* `delete` - (Defaults to 30 minutes) Used when deleting the Subscription Cost Management Export.

## Import

Subscription Cost Management Exports can be imported using the `resource id`, e.g.
```
